lua/neoclip/settings.lua

```diff
@@ -78,11 +78,19 @@ local function check_deprecated_entries()
 end
 
 local function setup(opts, subsettings)
+    --- `is_setable` returns `true` for dict-like and empty tables.
+    --- Thus, allowing to filter out array-like tables.
+    local function is_setable (t)
```

Comment:
[nit] rename to `is_dict`/`is_dict_like`?

could also live outside of `setup`? no need to be a closure?

Reply:
I am not familiar with Lua best practices. But assume that defining a function inside a recursive function is not a good idea in general, in terms of performance. (Probably it is not a problem for compiled languages:)). Though performance is not an issue here, I guess, I would keep the definition outside

I like `is_dict_like`
I will fix it tomorrow